test/nuts/pluginsGenerate.nut.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import * as os from 'os';
+import { execCmd } from '@salesforce/cli-plugins-testkit';
+import { expect } from 'chai';
+import * as shell from 'shelljs';
+
+let sfdxPluginDir: string;
+
+describe('plugins:generate', () => {
+  before(async () => {
+    sfdxPluginDir = path.join(os.tmpdir(), Date.now().toString(), '/sfdxPlugin');
+  });
+
+  it('plugins:generate generates a valid sfdx plugin', async () => {
+    const cmdOutput = execCmd(`plugins:generate --defaults ${sfdxPluginDir}`, {
+      ensureExitCode: 0,
+    });
+    expect(cmdOutput.shellOutput.code).equals(0);
+    expect(cmdOutput.shellOutput.stdout).includes('Created sfdxPlugin in');
+
+    const shellJsOpts = {
+      cwd: sfdxPluginDir,
+    };
+
+    const nyc = require.resolve('nyc/bin/nyc');
+    const mocha = require.resolve('mocha/bin/mocha');
```

Comment:
We need to resolve the module location and call these directly because `shell.exec` inheris the current env which yarn/mocha modifies and causes it to resolve to `sfdx-plugin-generate/node_modules/.bin` before `sfdxPluginDir/node_modules/.bin`